File: api/src/main/java/com/jlopez/bugtracker/model/BugPayload.java
package com.jlopez.bugtracker.model;

import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class BugPayload implements Serializable {

  private static final long serialVersionUID = -8865810957982887270L;

  private final Long id;

  private final String name;

  private final String description;

  private final StatusPayload status;

  private final LocalDateTime createdAt;

  private final LocalDateTime updatedAt;
}

File: client/src/shared/models/BugPayload.ts
import { StatusPayload } from './StatusPayload';

export class BugPayload {
    id: number | null;
    name: string;
    description: string;
    status: StatusPayload | null;
    createdAt: string;
    updatedAt: string;

    constructor() {
        this.id = null;
        this.name = '';
        this.description = '';
        this.status = null;
        this.createdAt = 'null';
        this.updatedAt = '';
    }

    static fromServerData(serverData: BugPayload): BugPayload {
        const newBug = new BugPayload();
        newBug.id = serverData.id;
        newBug.name = serverData.name;
        newBug.description = serverData.description;
        newBug.status = serverData.status;
        newBug.createdAt = serverData.createdAt;
        newBug.updatedAt = serverData.updatedAt;
        return newBug;
    }
}
